### DISCUSSION TITLE
Analytics/Telemetry/Error Tracking Options for Offline Usage

### DISCUSSION THREAD

**Discussion by lukethacoder** [tauri-apps/tauri#5959] [Q&A] [answered]
Just wondering what options are available for analytics/telemetry/error tracking within tauri applications. I understand any web-based 
 platform (GA, Sentry, MSFT Clarity etc.) should work fine when users have an active internet connection on the client side. You could probably fire off client-side events from rust too if you wanted here.

But when users are able to use your application offline and only occasionally connect to the internet is there a recommended way of handling this? I went down a bit of a rabbit hole of looking into more telemetry-based solutions (party sparked from https://github.com/tauri-apps/tauri/issues/166), but I think this would be a bit overkill for a couple of analytics events.

**Comment by JonasKruckenberg** [comment, marked as answer]
I'm only familiar with Sentry and know that their JavaScript SDK has support for offline caching, basically accumulating all reports locally if no internet connection could be established. I don't think the Rust SDK has support for this though, so you might only get partial offline support (and no offline caching if you use the sentry-tauri plugin because it routes everything through rust)

Edit: looking at the sentry rust docs a bit it seems like you could probably implement an offline-caching message queue by implementing the `Transport` trait yourselves

**Reply by lukethacoder** [reply]
Thanks Jonas. Will have a play around with the Sentry Rust SDK .
